kinto_client/__init__.py

```diff
@@ -118,12 +118,12 @@ def _paginated(self, endpoint, records=None, if_none_match=None, **kwargs):
                                        if_none_match=if_none_match)
         return records.values()
 
-    def _get_cache_headers(self, safe, data=None, last_modified=None):
+    def _get_cache_headers(self, safe, data=None, if_match=None):
         has_data = data is not None and data.get('last_modified')
-        if (last_modified is None and has_data):
-            last_modified = data['last_modified']
-        if safe and last_modified is not None:
-            return {'If-Match': utils.quote(last_modified)}
+        if (if_match is None and has_data):
+            if_match = data['last_modified']
+        if safe and if_match is not None:
+            return {'If-Match': utils.quote(if_match)}
```

Comment:
`utils.quote()` should be idempotent when the passed value is already a string with quotes (`"abc"`)